chacra/controllers/projects/binaries.py

```diff
@@ -65,30 +69,57 @@ def index(self):
     @secure(basic_auth)
     @index.when(method='POST', template='json')
     def index_post(self):
+        try:
+            data = request.json
+            name = data.get('name')
+        except ValueError:
+            error('/errors/invalid/', 'could not decode JSON body')
+
+        # updates the binary only if explicitly told to do so
+        if self.binary:
+            if not data.get('force'):
+                error('/errors/invalid/', 'file already exists and "force" flag was not used')
+            else:
+                # FIXME this looks like we need to implement PUT
+                path = data.get('path')
+                if path:
+                    try:
+                        data['size'] = os.path.getsize(path)
+                    except OSError:
+                        logger.exception('could not retrieve size from %s' % path)
+                        data['size'] = 0
+                self.binary.update_from_json(data)
+                return {}
+
+        # we allow empty data to be pushed
+        if not name:
+            error('/errors/invalid/', "could not find required key: 'name'")
+        name = data.pop('name')
+        path = data.get('path')
+
+        if path:
+            size = os.path.getsize(path)
+        else:
+            size = 0
+        Binary(
+            name=name, project=self.project, arch=self.arch,
+            distro=self.distro, distro_version=self.distro_version,
+            ref=self.ref, size=size
+        )
+
+        return {}
```

Comment:
I find it common to return that when exposing JSON otherwise the endpoint would return an empty body which can be awkward to handle if we are also returning JSON to report errors.